src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -579,7 +579,7 @@ private Conversion ClassifyExplicitBuiltInOnlyConversion(TypeSymbol source, Type
 
             if (HasUnboxingConversion(source, destination, ref useSiteDiagnostics))
             {
-                return Conversion.Unboxing;
+                return source.Kind == SymbolKind.TypeParameter ? Conversion.TypeParameterUnboxing : Conversion.ValueTypeUnboxing;
```

Comment:
Yes.  I think you're quite right.  VB has a whole host of narrowing type parameter conversions that are not being represented here.